### PR TITLE
This adds a polling mechanism to the framework

### PR DESCRIPTION
**Pull request by JackThomson2**
This polls the fitbit API for the users JSON.

### CODE REVIEW

File: src/main/java/beans/EnvriomentVariableBean.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+@Singleton(name = "EnvriomentVariableEJB")
+public class EnvriomentVariableBean {
+
+    // Fitbit Info
+    private String fitbitClientId;
+    private String fitbitClientSecret;
+    private String fitbitClientCallback;
+
+    //Aberfitness Info
+    private String aberfitnessClientId;
```
Comment (DavidFair) [inline src/main/java/beans/EnvriomentVariableBean.java]: These should all be:
`private final static string ABER_FITNESS_CLIENT_ID = System.getenv("fitbitClientId");`

As they do not change in the application runtime

File: src/main/java/datacollection/DataCheckThread.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    private Date now;
+
+    DataCheckThread(ConcurrentLinkedQueue<TokenMap> input, ConcurrentLinkedQueue<ProcessedData> out, OAuthBean oAuthBean) {
+        // Create Shallow copy to the global linked queue
+        this.input = input;
+        this.output = out;
+        this.oAuthBean = oAuthBean;
+
+        // Store the start of the requests
+        now = new Date();
```
Comment (DavidFair) [inline src/main/java/datacollection/DataCheckThread.java]: Just a note probably looking at `Instant` which is the replacement for date

File: src/main/java/beans/EnvriomentVariableBean.java (hunk 1 above)
Comment (JackThomson2) [reply src/main/java/beans/EnvriomentVariableBean.java]: Resolved in most recent commit

**Comment by DavidFair** [comment]
Please don't make massive PRs again. merging :smile:

**Pull request merged**